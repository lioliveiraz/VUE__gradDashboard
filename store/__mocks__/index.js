import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

export const getters = {
    isLoggedIn: jest.fn().mockReturnValue(true),
    getToken: jest.fn().mockReturnValue({ token: "token" }),
    getName: jest.fn().mockReturnValue(),
    getCourses: jest.fn().mockReturnValue([{ name: "course" }]),
    getAssessments: jest.fn().mockReturnValue([{ name: "assessments" }]),
    getScores: jest.fn().mockReturnValue([{ name: "scores" }])

};

export const mutations = {
    SET_TOKEN: jest.fn(),
    SET_ID: jest.fn(),
    SET_NAME: jest.fn(),
    SET_COURSES: jest.fn(),
    SET_ASSESSMENTS: jest.fn(),
    SET_SCORES: jest.fn(),
    ADD_COURSE: jest.fn(),
    ADD_ASSESSMENT: jest.fn(),
    ADD_SCORE: jest.fn()
};

export const actions = {
    login: jest.fn(),
    logout: jest.fn(),
    fetchCourses: jest.fn(),
    fetchScores: jest.fn(),
    handleAddCourse: jest.fn(),
    handleAddScore: jest.fn()


};

export const state = {
    token: "token",
    user_id: 1,
    user_name: "name",
    courses: [{ name: "course" }],
    assessments: [{ name: "assessments" }],
    scores: [{ name: "scores" }]
};

export function __createMocks(custom = { getters: {}, mutations: {}, actions: {}, state: {} }) {
    const mockGetters = Object.assign({}, getters, custom.getters);
    const mockMutations = Object.assign({}, mutations, custom.mutations);
    const mockActions = Object.assign({}, actions, custom.actions);
    const mockState = Object.assign({}, state, custom.state);

    return {
        getters: mockGetters,
        mutations: mockMutations,
        actions: mockActions,
        state: mockState,
        store: new Vuex.Store({
            modules: {
                auth: {
                    namespaced: true,
                    getters: mockGetters,
                    mutations: mockMutations,
                    actions: mockActions,
                    state: mockState,
                },
                courses: {
                    namespaced: true,
                    getters: mockGetters,
                    mutations: mockMutations,
                    actions: mockActions,
                    state: mockState,
                }
            }

        }),
    };
}

export const store = __createMocks().store;