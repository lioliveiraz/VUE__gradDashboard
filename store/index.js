import Vuex from 'vuex';
import Vue from 'vue';
import auth from './modules/auth';

Vue.use(Vuex);


const store = () => {
    return new Vuex.Store({
        modules: { auth }
    });
};

export default store;