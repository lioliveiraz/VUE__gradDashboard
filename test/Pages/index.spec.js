import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { __createMocks, store } from '../../store/__mocks__';
import Index from '../../pages';
import VueMeta from 'vue-meta';


jest.mock('../../store');

const localVue = createLocalVue();
localVue.use(VueMeta, { keyName: 'head' });
localVue.use(Vuex);

describe('<Index/>', () => {
    let wrapper;

    beforeEach(async () => {
        wrapper = await shallowMount(Index, {
            store: store,
            mocks: {
                $t: (msg) => msg
            },
            localVue
        });
    });
    afterEach(() => {
        if (wrapper) {
            wrapper.destroy();
        }
    });

    it('should render correctly', () => {
        const idArr = ["login_form", "login_img", "login_button"];
        idArr.forEach((id) => expect(wrapper.find(id)).toBeTruthy());
        expect(wrapper).toMatchSnapshot();

    });

    it('data should initialize correctly', () => {
        expect(Index.data().loginUserData).toBeTruthy();
        expect(Index.data().errors).toBeTruthy();

    });

    it('form should NOT  return error', async () => {
        const form = wrapper.find('form');
        await wrapper.setData({
            loginUserData: {
                empId: "123456",
                password: "123456"
            }
        });
        await form.trigger('submit');

        expect(wrapper.vm.errors).toStrictEqual({});
    });
    it("testing metaInfo", () => {
        expect(wrapper.vm.$metaInfo.title).toBe('Welcome');

    });
    it("should change the date once the method is called", async () => {
        await wrapper.vm.getUserInput("111111", "empId");
        let object = {
            empId: "111111"
        };
        expect(wrapper.vm.loginUserData).toEqual(expect.objectContaining(object));

    });


});