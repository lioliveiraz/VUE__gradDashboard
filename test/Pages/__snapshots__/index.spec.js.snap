// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index/> should render correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
